### PR TITLE
Move schema storage

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Move schema storage dir to /usr/share/susemanager/db out of the persistent container volume.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- Not needed

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23168

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+echo Going to reset pgsql database
+
+echo $PATH
+echo $PERLLIB
+
+export SYSTEMD_NO_WRAP=1
+#sysctl -w kernel.shmmax=18446744073709551615
+su - postgres -c "/usr/lib/postgresql/bin/pg_ctl stop" ||:
+su - postgres -c "/usr/lib/postgresql/bin/pg_ctl start" ||:
```
Comment (szachovy) [inline susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:24]: ?
```suggestion
su - postgres -c "/usr/lib/postgresql/bin/pg_ctl restart" ||:
```

**Review by szachovy — CHANGES_REQUESTED** [review]

**Review by rjmateus — APPROVED** [review]

**Review by mbussolotto — APPROVED** [review]
LGTM

File: python/spacewalk/spacewalk-backend.changes.mc.move-schema-storage [hunk 2]
```diff
@@ -0,0 +1 @@
+- schema dir moved to /usr/share/susemanager/db
```
Comment (szachovy) [inline python/spacewalk/spacewalk-backend.changes.mc.move-schema-storage:1]: Would it be possible to consider defining the path `/usr/share/susemanager/db` as a variable at the beginning of the script for better maintainability?

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
```
Comment (szachovy) [inline susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:3]: ?
```suggestion
set -euxo pipefail
```

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+export PERLLIB=/manager/spacewalk/setup/lib/:/manager/web/modules/rhn/:/manager/web/modules/pxt/:/manager/schema/spacewalk/lib
+export PATH=/manager/schema/spacewalk/:/manager/spacewalk/setup/bin/:$PATH
+
+echo Going to reset pgsql database
+
+echo $PATH
+echo $PERLLIB
+
+export SYSTEMD_NO_WRAP=1
```
Comment (szachovy) [inline susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:21]: I would like to suggest relocating all the `export` statements to the beginning of the file and refraining from using `echo` directly.

File: susemanager-utils/testing/docker/scripts/schema_migration_reportdb_test_pgsql.sh [hunk 5]
```diff
@@ -58,6 +58,16 @@ touch /var/lib/rhn/rhn-satellite-prep/etc/rhn/rhn.conf
...
 smdba system-check autotuning --max_connections=50
 
+
+# we changed the schema dir, but we start with a schema which live still in the old location
+# provide a symlink to make the tooling work
+if [ -d /etc/sysconfig/rhn/postgres -a ! -e /usr/share/susemanager/db/postgres ]; then
+	mkdir -p /usr/share/susemanager/db
+	ln -s /etc/sysconfig/rhn/postgres /usr/share/susemanager/db/postgres
+	ln -s /etc/sysconfig/rhn/reportdb /usr/share/susemanager/db/reportdb
+fi
```
Comment (szachovy) [inline susemanager-utils/testing/docker/scripts/schema_migration_reportdb_test_pgsql.sh:68]:
```suggestion
if [ -d /etc/sysconfig/rhn/postgres -a ! -e /usr/share/susemanager/db/postgres ]; then
    mkdir -p /usr/share/susemanager/db
    ln -s /etc/sysconfig/rhn/postgres /usr/share/susemanager/db/postgres
    ln -s /etc/sysconfig/rhn/reportdb /usr/share/susemanager/db/reportdb
fi
```

File: susemanager-utils/testing/docker/scripts/schema_migration_test_pgsql.sh [hunk 6]
```diff
@@ -58,6 +58,13 @@ touch /var/lib/rhn/rhn-satellite-prep/etc/rhn/rhn.conf
 cp /root/rhn.conf /etc/rhn/rhn.conf
 smdba system-check autotuning --max_connections=50
 
+# we changed the schema dir, but we start with a schema which live still in the old location
+# provide a symlink to make the tooling work
+if [ -d /etc/sysconfig/rhn/postgres -a ! -e /usr/share/susemanager/db/postgres ]; then
+        mkdir -p /usr/share/susemanager/db
+        ln -s /etc/sysconfig/rhn/postgres /usr/share/susemanager/db/postgres
+fi
```
Comment (szachovy) [inline susemanager-utils/testing/docker/scripts/schema_migration_test_pgsql.sh:66]:
```suggestion
if [ -d /etc/sysconfig/rhn/postgres -a ! -e /usr/share/susemanager/db/postgres ]; then
    mkdir -p /usr/share/susemanager/db
    ln -s /etc/sysconfig/rhn/postgres /usr/share/susemanager/db/postgres
fi
```

File: testsuite/podman_runner/07_start_server.sh [hunk 7]
```diff
@@ -2,5 +2,5 @@
...
+sudo -i podman run --rm --tmpfs /run -v ${src_dir}/schema/spacewalk/spacewalk-schema-upgrade:/usr/bin/spacewalk-schema-upgrade -v${src_dir}/testsuite:/testsuite -v ${src_dir}/schema/spacewalk/upgrade/:/usr/share/susemanager/db/schema-upgrade/ -v ${src_dir}/schema/reportdb/upgrade/:/usr/share/susemanager/db/reportdb-schema-upgrade/ -v ${src_dir}/web:/web -v ${src_dir}/branding:/branding -v ${src_dir}/java:/java -v ${src_dir}/client:/client -v /sys/fs/cgroup:/sys/fs/cgroup:rw -v /tmp/test-all-in-one:/tmp --cgroupns=host -h uyuni-server-all-in-one-test -p 8443:443 -p 8080:80 -p 4505:4505 -p 4506:4506 -d --name=uyuni-server-all-in-one-test --network uyuni-network-1 ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-server-all-in-one-dev:$UYUNI_VERSION
```
Comment (szachovy) [inline testsuite/podman_runner/07_start_server.sh:5]: Would it be more readable to have it like that?: _(needs verification)_
```suggestion
sudo -i podman run --rm \
    --tmpfs /run \
    -v ${src_dir}/schema/spacewalk/spacewalk-schema-upgrade:/usr/bin/spacewalk-schema-upgrade \
    -v ${src_dir}/testsuite:/testsuite \
    -v ${src_dir}/schema/spacewalk/upgrade/:/usr/share/susemanager/db/schema-upgrade/ \
    -v ${src_dir}/schema/reportdb/upgrade/:/usr/share/susemanager/db/reportdb-schema-upgrade/ \
    -v ${src_dir}/web:/web \
    -v ${src_dir}/branding:/branding \
    -v ${src_dir}/java:/java \
    -v ${src_dir}/client:/client \
    -v /sys/fs/cgroup:/sys/fs/cgroup:rw \
    -v /tmp/test-all-in-one:/tmp \
    --cgroupns=host \
    -h uyuni-server-all-in-one-test \
    -p 8443:443 \
    -p 8080:80 \
    -p 4505:4505 \
    -p 4506:4506 \
    -d --name=uyuni-server-all-in-one-test \
    --network uyuni-network-1 \
    ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-server-all-in-one-dev:$UYUNI_VERSION
```

File: python/spacewalk/spacewalk-backend.changes.mc.move-schema-storage (hunk 2 above)
Comment (mcalmer) [reply python/spacewalk/spacewalk-backend.changes.mc.move-schema-storage:1]: no, too many different packages and languages. And the code for reading a config file is just too complicated for a 1 time usage. Additionally we do not plan to change this again. The old location was there for more than 20 years.

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh (hunk 3 above)
Comment (mcalmer) [reply susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:3]: Not in this PR. Feel free to start an own.

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh (hunk 4 above)
Comment (mcalmer) [reply susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:21]: Not in this PR. Feel free to start an own

**Review by srbarrios — APPROVED** [review]
I only checked podman_runner, that part LGTM.

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh (hunk 3 above)
Comment (szachovy) [reply susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:3]: ok

File: python/spacewalk/spacewalk-backend.changes.mc.move-schema-storage (hunk 2 above)
Comment (szachovy) [reply python/spacewalk/spacewalk-backend.changes.mc.move-schema-storage:1]: ok

File: susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh (hunk 4 above)
Comment (szachovy) [reply susemanager-utils/testing/docker/scripts/init-pgsql-reportdb4eclipse.sh:21]: ok

**Review by szachovy — APPROVED** [review]

**Review by cbosdo — APPROVED** [review]

**Comment by mcalmer** [comment]
@uyuni-project/release-engineering As soon as Alex has reviewed it, it can be merged when you think it is a good time for doing it.

**Review by agraul — APPROVED** [review]

**Pull request merged**